sql/core/src/main/scala/org/apache/spark/sql/scripting/SqlScriptingExecutionNode.scala

```diff
@@ -119,20 +148,112 @@ class SingleStatementExec(
     origin.sqlText.get.substring(origin.startIndex.get, origin.stopIndex.get + 1)
   }
 
-  override def reset(): Unit = isExecuted = false
+  override def reset(): Unit = {
+    raisedError = false
+    errorState = None
+    error = None
+    rethrow = None
+    result = None // Should we do this?
+  }
+
+  def execute(session: SparkSession): Unit = {
+    try {
+      val rows = Some(Dataset.ofRows(session, parsedPlan).collect())
+      if (shouldCollectResult) {
+        result = rows
+      }
+    } catch {
+      case e: SparkThrowable =>
+        raisedError = true
+        errorState = Some(e.getSqlState)
+        error = Some(e)
+        e match {
+          case throwable: Throwable =>
```

Comment:
@cloud-fan Are there any cases where something that implements `SparkThrowable` isn't also `Throwable`?